server/user.js

```diff
@@ -0,0 +1,85 @@
+const { TimeLogger } = require("../src/util");
+const { R } = require("redbean-node");
+const { UptimeKumaServer } = require("./uptime-kuma-server");
+const server = UptimeKumaServer.getInstance();
+const io = server.io;
+
+/**
+ * Send list of users to client
+ * @param {Socket} socket Socket.io socket instance
+ * @returns {Promise<Bean[]>}
+ */
+async function sendUserList(socket) {
+    const timeLogger = new TimeLogger();
+
+    let result = [];
+    let list = (await R.findAll("user")).map(({ id, username, active }) => ({
+        id,
+        username,
+        active
+    }));
+
+    io.to(socket.userID).emit("userList", result);
+
+    timeLogger.print("Send User List");
+
+    return list;
+}
+
+/**
+ * Send list of users to client
+ * @param {Socket} socket Socket.io socket instance
+ * @returns {Promise<Bean[]>}
+ */
+async function getUser(userID) {
+    const timeLogger = new TimeLogger();
+
+    const bean = await R.findOne("user", " id = ? ", [ userID ])
+        .then(({ id, username, active }) => ({
+            id,
+            username,
+            active
+        }));
+
+    if (!bean) {
+        throw new Error("User not found");
+    }
+
+    timeLogger.print(`Get user ${userID}`);
+
+    return bean;
```

Comment:
why the return statement? (it is not being used where the function is called from)


```suggestion
```

Reply:
You are right it's useless and don't return it allow me to remove the `.map` used to keep control over returned fields. In fact I'm usually writing GraphQL API, where I have to return the updated object as mutation result (used by the webapp to refresh his cache).

Reply:
That return statement is for `getUser` function, which is used in `server.js` at L.654:
```js
        socket.on("getUser", async (userID, callback) => {
            try {
                checkLogin(socket);

                const user = await getUser(userID);

                callback({
                    ok: true,
                    user
                });
            } catch (e) {
                callback({
                    ok: false,
                    msg: e.message,
                });
            }
        });
```
So I absolutely need to return the `user` here!

Reply:
sorry, I must have somehow not found it during the code-review (idk, how tbh)

Reply:
no pb 😘